docs/psoc6/quickref.rst

```diff
@@ -347,3 +347,83 @@ The constructor can be called by passing the required arguments. All initializat
     pwm.init(freq=90,duty_us=100,invert=1)    # Modify the settings of PWM object
     pwm.deinit()                              # Deinitialisation of PWM pin
 
+Software SPI bus
+----------------
+
+Software SPI (using bit-banging) works on all pins, and is accessed via the
+:ref:`machine.SoftSPI <machine.SoftSPI>` class::
+
+    from machine import Pin, SoftSPI
+
+    # construct a SoftSPI bus on the given pins
+    # polarity is the idle state of SCK
+    # phase=0 means sample on the first edge of SCK, phase=1 means the second edge
+    spi = SoftSPI(baudrate=100_000, polarity=1, phase=0, sck='P0_2', mosi='P0_0', miso='P0_1')
+
+    spi.init(baudrate=200000) # set the baudrate
+
+    spi.read(10)            # read 10 bytes on MISO
+    spi.read(10, 0xff)      # read 10 bytes while outputting 0xff on MOSI
+
+    buf = bytearray(50)     # create a buffer
+    spi.readinto(buf)       # read into the given buffer (reads 50 bytes in this case)
+    spi.readinto(buf, 0xff) # read into the given buffer and output 0xff on MOSI
+
+    spi.write(b'12345')     # write 5 bytes on MOSI
+
+    buf = bytearray(4)      # create a buffer
+    spi.write_readinto(b'1234', buf) # write to MOSI and read from MISO into the buffer
+    spi.write_readinto(buf, buf) # write buf to MOSI and read MISO back into buf
+
+.. Warning::
+   Currently *all* of ``sck``, ``mosi`` and ``miso`` *must* be specified when
+   initialising Software SPI.
+
+Hardware SPI bus
+----------------
+Hardware SPI works on the following listed pair of SPI pins. 
+
+=====  ===========  ============  ============
+\      Default  
+=====  ===========  ============  ============
+MOSI    P9_0           P6_0         P10_0
+MISO    P9_1           P6_1         P10_1
+SCK     P9_2           P6_2         P10_2
+=====  ===========  ============  ============
+
+Refer `PSoC 6 MCU: CY8C62x8, CY8C62xA Datasheet <https://www.infineon.com/dgdl/Infineon-PSOC_6_MCU_CY8C62X8_CY8C62XA-DataSheet-v18_00-EN.pdf?fileId=8ac78c8c7d0d8da4017d0ee7d03a70b1>`_
+for additional details regarding all the SPI capable pins. 
+
+The driver is accessed via :ref:`machine.SPI <machine.SPI>`
+
+The constructor
+^^^^^^^^^^^^^^^
+An instance of the :mod:`machine.SPI` class can be created by invoking the constructor with all the 
+necessary parameters to fully configure and initialise the ``SPI``. By invoking the constructor with no additional parameters 
+SPI object is created with default settings or settings of previous initialisation if any. 
+
+::
+    
+    from machine import SPI
+    spi = SPI(0) # Default assignment: id=0, sck=P9_2  ,MOSI=P9_0, MISO=P9_1, baudrate=1000000, Polarity=0, Phase=0, bits=8, firstbit=SPI.MSB
+    spi.init()
+
+Management of a CS signal should happen in user code (via machine.Pin class).
+
+::
+    from machine import Pin
+    cs = Pin('P9_3', mode=Pin.OUT, value=1)      # Create chip-select on pin P9_3
+    cs(0)                                        # select the peripheral
```

Comment:
I do not understand this command : cs(0) on a pin object. What does this do ?

Reply:
This is the chip select pin which is an active low signal. So by cs(0) it will select the pin corresponding pin & makes it value=0.

Reply:
Why do you not set it to 0 in the line above ?

Reply:
https://docs.micropython.org/en/latest/library/machine.SPI.html 
Here in the common example given by micropython, it's like that. So in order to follow the same I did that.

Reply:
It makes sense to initialize it high (not selected), and the select the device later pulling the line low no?
It makes sense how is implemented by @IFX-Anusha.
But of course, you could directly initialize it low, if you directly intent to select that device. But this is more illustrative I feel.